### ISSUE TITLE
Dual Role keys [feature request]

### ISSUE THREAD

**Issue by toneman77** [DavRack/skr#4]
A dual-role feature would be awesome. (see: [here](https://en.wikipedia.org/wiki/Modifier_key#Dual-role_keys))
For example I have the CapsLock key mapped to a dual-role key on my programmable keyboard:
- if you tap caps lock --> Esc
- if you hold caps lock --> Ctrl

Other ppl also used
- SpaceFN (spacekey: tap=space, hold=layerswitch)
- [space cadet shift](https://stevelosh.com/blog/2012/10/a-modern-space-cadet/#s15-better-shifting) (left and right shift keys: tap = parenthesis open/close, hold=shift)

**Comment by DavRack** [comment]
Yeah, I agree, that would be awesome! One of the projects (see projects tab on github) is "Dual function fnkey" so that feature its on the way.

**Comment by kflak** [comment]
Hi there, and thanks for an awesome tool! 

I would also love to see this. I've been running the caps2esc plugin to use the caps key as mentioned in OP's post. However, it seems not possible to run both this _and_ skr at the same time. Or is there? If so, I would be OK running both services, but it would obviously be better to have all my keyboard mods in the same service...
